### PR TITLE
Fix a small bug in the attention bias calculation when flash attention is not available

### PR DESCRIPTION
**Pull request by tbuthfer**
Fix a small bug in the attention bias calculation when flash attention is not available.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -64,13 +64,21 @@ def forward(self, x):
...
+
+            # causal means the attention per token is not affected by future tokens, achieved by adding a bias
+            attn_bias = torch.zeros(k.size(-2), k.size(-2), dtype=q.dtype)
+            temp_mask = torch.ones(k.size(-2), k.size(-2), dtype=torch.bool).tril()
+            attn_bias.masked_fill_(temp_mask.logical_not(), float("-inf"))
+            att += attn_bias
+
             att = F.softmax(att, dim=-1)
-            att = self.attn_dropout(att)
+            att = self.attn_dropout(att) if self.training else att
```
Comment (vhmth) [inline model.py:77]: I believe this shouldn't be necessary since `nn.Dropout` will not apply dropout if `self.training = True`.

Dropout source: https://pytorch.org/docs/stable/_modules/torch/nn/modules/dropout.html#Dropout

Which calls `F.dropout`:

https://pytorch.org/docs/stable/generated/torch.nn.functional.dropout.html

Notice it takes a `training` arg which comes from `nn.Dropout`'s `self.training` state.

**Review by vhmth — CHANGES_REQUESTED** [review]
